gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/DefaultTokenStateService.java

```diff
@@ -237,12 +249,37 @@ protected boolean isUnknown(final String token) {
 
   protected void updateExpiration(final String tokenId, long expiration) {
     synchronized (tokenExpirations) {
-      tokenExpirations.replace(tokenId, expiration);
+      if (tokenExpirations.containsKey(tokenId)) {
+        tokenExpirations.replace(tokenId, expiration);
+      } else {
+        tokenExpirations.put(tokenId, expiration);
+      }
```

Comment:
`Map.replace` also runs a `containsKey`:
```
The default implementation is equivalent to, for this map:
 
 if (map.containsKey(key)) {
     return map.put(key, value);
 } else
     return null;
```
I think a simple `put` is enough: you want to make sure you have the given expiration in `tokenExpirations`

Reply:
Yes, I had meant to return to this after verifying that the replace invocation that was already there is not needed.